applications/nrf5340_audio/src/bluetooth/bt_management/advertising/bt_mgmt_adv.c

```diff
@@ -73,11 +73,59 @@ static void bond_find(const struct bt_bond_info *info, void *user_data)
 	}
 }
 
+/**
+ * @brief Prints the address of the local device and the remote device.
+ * (If directed advertisement is active).
+ */
+static int addr_print(bt_addr_le_t const *const local_addr, bt_addr_le_t const *const dir_adv_addr)
+{
+	char local_addr_str[BT_ADDR_LE_STR_LEN] = {'\0'};
+	char directed_to_addr_str[BT_ADDR_LE_STR_LEN] = {'\0'};
+
+	if (local_addr == NULL) {
+		return -EINVAL;
+	}
+
+	(void)bt_addr_le_to_str(local_addr, local_addr_str, BT_ADDR_LE_STR_LEN);
+	LOG_INF("Local addr: %s", local_addr_str);
+
+	if (dir_adv_addr != NULL) {
+		(void)bt_addr_le_to_str(dir_adv_addr, directed_to_addr_str, BT_ADDR_LE_STR_LEN);
+		LOG_INF("Adv directed to: %s.", directed_to_addr_str);
+	}
+
+	return 0;
+}
+
+#if defined(CONFIG_BT_PRIVACY)
+static bool adv_rpa_expired_cb(struct bt_le_ext_adv *adv)
+{
+	int ret;
+	struct bt_le_ext_adv_info ext_adv_info;
+
+	LOG_INF("RPA (Resolvable Private Address) expired.");
+
+	ret = bt_le_ext_adv_get_info(ext_adv, &ext_adv_info);
+	ERR_CHK_MSG(ret, "bt_le_ext_adv_get_info failed");
```

Comment:
I see that all _ASSERTs have been changed to ERR_CHK_MSG, but I think we shouldn't fail the application if it's failing to print this address. A LOG_WRN or LOG_ERR should be enough.

Reply:
Now it will fail on the get_info and only print if addr_print fails.